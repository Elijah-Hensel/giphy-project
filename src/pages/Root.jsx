import { useState, useEffect } from "react"
import { useSearchParams } from "react-router-dom"
import SearchForm from "../components/SearchForm"
import GifList from "../components/GifList"
// import { setGifsByQuery, setInitialGifs } from "../utils/actions"
import { getInitialGifs, getGifsBySearchQuery } from "../api"

function Root() {
  const [params] = useSearchParams()
  const [gifs, setGifs] = useState({
    data: [],
    pagination: null,
  })

  useEffect(() => {
    const query = params.get("q")
    if (query) {
      const setGifsByQuery = async (query, action) => {
        console.log({ query, action }, "YEEEEEEHAW")
        const { data, pagination } = await getGifsBySearchQuery(query)
        action({ data, pagination })
      }
      setGifsByQuery(query, setGifs).catch((err) => console.error(err))
    } else {
      const setInitialGifs = async (action) => {
        const { data, pagination } = await getInitialGifs()
        action({ data, pagination })
      }
      setInitialGifs(setGifs).catch((err) => console.error(err))
    }
  }, [params])

  return (
    <div className="Root">
      <SearchForm setGifs={setGifs} />
        <GifList data-testid="gif-list" gifData={gifs} />
    </div>
  )
}

export default Root
