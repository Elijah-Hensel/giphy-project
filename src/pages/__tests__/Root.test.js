import React from 'react'
import { render, act, waitFor, fireEvent } from '@testing-library/react'
import Root from '../Root'
import { BrowserRouter } from 'react-router-dom'

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'), // use actual for all non-hook parts
  useParams: () => ({
    q: "big"
  }),
}));

describe('Root', () => {
  it('renders without crashing', async () => {
    const { container } = render(<Root />, {wrapper: BrowserRouter})
    await waitFor(() => container.querySelector('video'))
    expect(container).toBeDefined()
  })

  // it('displays gifs from the API', async () => {
  //   const { queryByTestId } = render(<Root />, {wrapper: BrowserRouter})
  //   await waitFor(() => queryByTestId('gif-list'))
  //   expect(queryByTestId('gif-list')).toBeDefined()
  // })

  // it('searches for gifs', async () => {
  //   const { queryByTestId, getByPlaceholderText, queryByDisplayValue } = render(<Root />, {wrapper: BrowserRouter})
  //   // fireEvent.click(queryByTestId('search-button'))
  //   const input = getByPlaceholderText('What kind of GIF would you like to see..?')
  //   act(() => {
  //     fireEvent.change(input, { target: { value: 'big' } })
  //   })
  //   fireEvent.click(queryByTestId('search-button'))
  //   await waitFor(() => queryByTestId('gif-list'))

  //   const gifs = document.querySelectorAll('video')
  //   // const gifsDescriptions = gifs.map((gif) => gif.value)
  //   console.log({gifs})
  //   // expect(gifsDescriptions).toEqual(gifsDescriptions)
  // })
})
