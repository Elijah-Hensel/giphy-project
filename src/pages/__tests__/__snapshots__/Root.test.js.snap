// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Root renders without crashing 1`] = `
<div>
  <div
    class="Root"
  >
    <div
      class="search-form-container"
    >
      <form
        id="search-form"
      >
        <input
          class="search-form-input"
          name="search-params"
          placeholder="What kind of GIF would you like to see..?"
          type="text"
          value=""
        />
        <button
          class="btn-primary"
          disabled=""
          type="submit"
          value="submit"
        >
          Search!
        </button>
      </form>
    </div>
    <div />
  </div>
</div>
`;
